sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2239,6 +2239,34 @@ class Dataset[T] private[sql](
     }
   }
 
+   /**
+    * Returns a new Dataset with altered column names.
+    * This is a no-op if schema doesn't contain existingName.
+    *
+    * @param convert conversion function from the old to the new name
+    *
+    * @group untypedrel
+    * @since 2.0.0
+    */
+  def withAllColumnsRenamed(convert: String => String): DataFrame = {
+    val output = queryExecution.analyzed.output
+    var containsRename = false
```

Comment:
This should not be optimised for the case when convert function is the identity.  This way you could get rid of the flag 'containsRename'.

Reply:
Thats correct. Removed that optimization and the code became much smaller.